src/paho/mqtt/client.py

```diff
@@ -625,6 +624,196 @@ def __init__(self, client_id="", clean_session=None, userdata=None,
+
+    @property
+    def protocol(self) -> int:
+        "Protocol version used (MQTT v3, MQTT v3.11, MQTTv5)"
+        return self.protocol
+
+    @property
+    def connect_timeout(self) -> float:
+        "Timeout used to establish the TCP (& TLS / websocket if enabled) in seconds"
+        return self._connect_timeout
+
+    @connect_timeout.setter
+    def connect_timeout(self, value: float):
+        "Change connect_timeout for future (re)connection"
+        if value <= 0.0:
+            raise ValueError("timeout must be a positive number")
+
+        self._connect_timeout = value
+
+    @property
+    def username(self) -> str | None:
+        "Return the username use to connect to MQTT broken or None if not credenials are used."
+        if self._username is None:
+            return None
+        return self._username.decode("utf-8")
+
+    @username.setter
+    def username(self, value: str | None) -> None:
+        """
+        Update username. This will only be used on future (re)connection. You should probably
+        use reconnect() to update the connection if established.
+        """
+        if value is None:
+            self._username = None
+        else:
+            self._username = value.encode("utf-8")
+
+    @property
+    def password(self) -> str | None:
+        "Return the password use to connect to MQTT broken or None if not password are used."
+        if self._password is None:
+            return None
+        return self._password.decode("utf-8")
+
+    @password.setter
+    def password(self, value: str | None) -> None:
+        """
+        Update password. This will only be used on future (re)connection. You should probably
+        use reconnect() to update the connection if established.
+        """
+        if value is None:
+            self._password = None
+        else:
+            self._password = value.encode("utf-8")
+
+    @property
+    def max_inflight_messages(self) -> int:
+        "Maximum number of message with QoS > 0 that can be part way through their network flow at once"
+        return self._max_inflight_messages
+
+    @max_inflight_messages.setter
+    def max_inflight_messages(self, value: int) -> None:
+        "Update max_inflight_messages. It's behavior is undefined if the connection is already open"
+        if self._sock is not None:
+            # Not tested. Some doubt that everything is okay when max_inflight change between 0
+            # and > 0 value because _update_inflight is skipped when _max_inflight_messages == 0
+            warnings.warn(
+                "updating max_inflight_messages on established connection is not supported",
+                stacklevel=2,
+            )
+
+        if value < 0:
+            raise ValueError("Invalid inflight.")
+
+        self._max_inflight_messages = value
+
+    @property
+    def max_queued_messages(self) -> int:
+        "Maximum number of message with QoS > 0 that can be part way through their network flow at once"
```

Comment:
```suggestion
        """Maximum number of messages with QoS > 0 that can be partway through the network flow at once"""
```
... same concern about "partway through the network flow" – and then, this string is exactly the same as `max_inflight_messages`, so I'm not sure what the actual difference is? I should hope there is a difference.

Reply:
Here it's a copy/paste mistake I made. Should the same description of max_queued_messages_set which don't imply network, only internal queue.